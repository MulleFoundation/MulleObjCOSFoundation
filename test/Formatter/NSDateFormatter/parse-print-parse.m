//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#ifdef __MULLE_OBJC__
# import <MulleObjCOSFoundation/MulleObjCOSFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif


static void   test_formatter( NSDateFormatter *formatter,
                              NSString *dateString,
                              int depth)
{
   NSDate     *date;
   NSString   *s;

   date = [formatter dateFromString:dateString];
   s    = [formatter stringFromDate:date];

   printf( "   %s: %s (%0.17g)\n",
               [NSStringFromClass( [date class]) UTF8String],
               [s UTF8String],
               [date timeIntervalSinceReferenceDate]);

   if( ! depth)
      test_formatter( formatter, s, 1);
}


static void   test( NSDateFormatter *formatter, NSString *dateString)
{
   printf( "%s\n", [dateString UTF8String]);

   [formatter setGeneratesCalendarDates:NO];
   test_formatter( formatter, dateString, 0);

   [formatter setGeneratesCalendarDates:YES];
   test_formatter( formatter, dateString, 0);
}


int   main( int argc, const char * argv[])
{
   NSDateFormatter   *formatter;

#ifdef __MULLE_OBJC__
   if( mulle_objc_global_check_universe( __MULLE_OBJC_UNIVERSENAME__) != mulle_objc_universe_is_ok)
      return( 1);
#endif

   formatter = [[[NSDateFormatter alloc] initWithDateFormat:@"%Y-%m-%dT%H:%M:%S:%F%z"
                                       allowNaturalLanguage:NO] autorelease];

   test( formatter, @"1970-01-01T00:00:00:000+0000");
   test( formatter, @"1970-01-01T00:00:00:000+0300");

   test( formatter, @"2013-12-11T10:09:08:765+0000");
   test( formatter, @"2013-12-11T10:09:08:765+01:00");
   // some wacky timezone
   test( formatter, @"2013-02-24T20:08:15:123+0845");

   return( 0);
}
