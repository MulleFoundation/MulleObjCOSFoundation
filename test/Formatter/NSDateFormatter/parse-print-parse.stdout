1970-01-01T00:00:00:000+0000
   _MulleObjCConcreteDate: 1970-01-01T00:00:00:000Z (-978307200)
   _MulleObjCConcreteDate: 1970-01-01T00:00:00:000Z (-978307200)
   _MulleObjCConcreteCalendarDate: 1970-01-01T00:00:00:000Z (-978307200)
   _MulleObjCConcreteCalendarDate: 1970-01-01T00:00:00:000Z (-978307200)
1970-01-01T00:00:00:000+0300
   _MulleObjCConcreteDate: 1969-12-31T21:00:00:000Z (-978318000)
   _MulleObjCConcreteDate: 1969-12-31T21:00:00:000Z (-978318000)
   _MulleObjCConcreteCalendarDate: 1970-01-01T00:00:00:000+0300 (-978318000)
   _MulleObjCConcreteCalendarDate: 1970-01-01T00:00:00:000+0300 (-978318000)
2013-12-11T10:09:08:765+0000
   _MulleObjCConcreteDate: 2013-12-11T10:09:08:765Z (408449348.7650001)
   _MulleObjCConcreteDate: 2013-12-11T10:09:08:765Z (408449348.7650001)
   _MulleObjCConcreteCalendarDate: 2013-12-11T10:09:08:000Z (408449348)
   _MulleObjCConcreteCalendarDate: 2013-12-11T10:09:08:000Z (408449348)
2013-12-11T10:09:08:765+01:00
   _MulleObjCConcreteDate: 2013-12-11T09:09:08:765Z (408445748.7650001)
   _MulleObjCConcreteDate: 2013-12-11T09:09:08:765Z (408445748.7650001)
   _MulleObjCConcreteCalendarDate: 2013-12-11T10:09:08:000+0100 (408445748)
   _MulleObjCConcreteCalendarDate: 2013-12-11T10:09:08:000+0100 (408445748)
2013-02-24T20:08:15:123+0845
   _MulleObjCConcreteDate: 2013-02-24T11:23:15:123Z (383397795.12299991)
   _MulleObjCConcreteDate: 2013-02-24T11:23:15:123Z (383397795.12299991)
   _MulleObjCConcreteCalendarDate: 2013-02-24T20:08:15:000+0845 (383397795)
   _MulleObjCConcreteCalendarDate: 2013-02-24T20:08:15:000+0845 (383397795)
